### ISSUE TITLE
Verifications not functioning in latest release

### ISSUE THREAD

**Issue by tpickett66** [jwt/ruby-jwt#128]
After updating an app to `1.5.2` we're seeing some test failures due to exceptions not being raised when issuer, audience, and subject claims are mismatched. I'll investigate further and add info as I find it.

**Comment by tpickett66** [comment]
Haha, after trying to reproduce the issue outside the app I'm not seeing it. Sorry for the false alarm.

**Comment by tpickett66** [comment]
As it turns out there was a change. The verification routines still work but only when the `verify_*` key in the options hash is a symbol.

```ruby
require 'jwt'

key = 'superSekretK3yz!'
config = {
  'iss' => 'foo'
}

payload = {
  foo: 'bar',
}

token = JWT.encode(config.merge(payload), key)
decoded = JWT.decode(token, key, true, {'iss' => 'other', 'verify_iss' => true}) # succeeds in decoding
decoded = JWT.decode(token, key, true, {'iss' => 'other', verify_iss: true}) # fails to decode
```

**Comment by tpickett66** [comment]
Looks using the above as the basis of a good/bad script for `git bisect` I landed on c490860b1281575d0fcf8ba342b3889f18d770d1 as the first commit with the changed behavior. Based on the commit message `Change hash syntax` it looks like the change was intentional :-(

**Comment by excpt** [comment]
Thanks for the feedback and your investigation results.
This is a nice solution for the string vs. hash situation.
Thanks for the contribution.
